### ISSUE TITLE
GGUI Not Available

### ISSUE THREAD

**Issue by aespielberg** [taichi-dev/taichi#5056]
**Describe the bug**
GGUI not available when running example1 in taichi voxel challenge

**To Reproduce**
Run example1.py in taichi voxel challenge

**Log/Screenshots**
```
$ python mpm3d.py 
[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, linux, python 3.9.12
[Taichi] Starting on arch=cuda
65536
Loading presets...this might take a minute
Traceback (most recent call last):
  File "/home/aespielberg/ResearchCode/taichi_voxel/mpm3d.py", line 253, in <module>
    window = ti.ui.Window("Real MPM 3D", res, vsync=False)
  File "/home/aespielberg/anaconda3/envs/taichi_voxel/lib/python3.9/site-packages/taichi/ui/window.py", line 22, in __init__
    check_ggui_availability()
  File "/home/aespielberg/anaconda3/envs/taichi_voxel/lib/python3.9/site-packages/taichi/ui/utils.py", line 71, in check_ggui_availability
    raise Exception("GGUI Not Available")
Exception: GGUI Not Available

```
**Additional comments**
```
Here are the results of ti diagnose:

$ ti diagnose
[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, linux, python 3.9.12

*******************************************
**      Taichi Programming Language      **
*******************************************

Docs:   https://docs.taichi-lang.org/
GitHub: https://github.com/taichi-dev/taichi/
Forum:  https://forum.taichi.graphics/

Taichi system diagnose:

python: 3.9.12 (main, Apr  5 2022, 06:56:58) 
[GCC 7.5.0]
system: linux
executable: /home/aespielberg/anaconda3/envs/taichi_voxel/bin/python
platform: Linux-4.15.0-142-generic-x86_64-with-glibc2.23
architecture: 64bit ELF
uname: uname_result(system='Linux', node='AuroraUnit', release='4.15.0-142-generic', version='#146~16.04.1-Ubuntu SMP Tue Apr 13 09:27:15 UTC 2021', machine='x86_64')
locale: en_US.UTF-8
PATH: /home/aespielberg/gcloud/google-cloud-sdk/bin:/home/aespielberg/anaconda3/envs/taichi_voxel/bin:/home/aespielberg/anaconda3/condabin:/home/aespielberg/bin:/home/aespielberg/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
PYTHONPATH: ['/home/aespielberg/anaconda3/envs/taichi_voxel/bin', '/home/aespielberg/ResearchCode/taichi_voxel', '/home/aespielberg/anaconda3/envs/taichi_voxel/lib/python39.zip', '/home/aespielberg/anaconda3/envs/taichi_voxel/lib/python3.9', '/home/aespielberg/anaconda3/envs/taichi_voxel/lib/python3.9/lib-dynload', '/home/aespielberg/anaconda3/envs/taichi_voxel/lib/python3.9/site-packages']

No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial



import: <module 'taichi' from '/home/aespielberg/anaconda3/envs/taichi_voxel/lib/python3.9/site-packages/taichi/__init__.py'>

cc: False
cpu: True
metal: False
opengl: False
cuda: True
vulkan: False

`glewinfo` not available: [Errno 2] No such file or directory: 'glewinfo'

Sun May 29 03:08:10 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.56       Driver Version: 418.56       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1070    Off  | 00000000:01:00.0  On |                  N/A |
| N/A   56C    P8    12W /  N/A |    284MiB /  8117MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1304      G   /usr/lib/xorg/Xorg                           172MiB |
|    0      3000      G   compiz                                        42MiB |
|    0      3341      G   ...uest-channel-token=10820157209923482346    36MiB |
|    0      3438      G   ...quest-channel-token=6647002422489032078    30MiB |
+-----------------------------------------------------------------------------+

[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, linux, python 3.9.12

[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, linux, python 3.9.12
[Taichi] Starting on arch=x64

[W 05/29/22 03:08:11.447 5286] [misc.py:adaptive_arch_select@781] Arch=[<Arch.opengl: 7>] is not supported, falling back to CPU
[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, linux, python 3.9.12
[Taichi] Starting on arch=x64

[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, linux, python 3.9.12
[Taichi] Starting on arch=cuda

[Taichi] version 1.0.2, llvm 10.0.0, commit 14f0acba, linux, python 3.9.12

*******************************************
**      Taichi Programming Language      **
*******************************************

Docs:   https://docs.taichi-lang.org/
GitHub: https://github.com/taichi-dev/taichi/
Forum:  https://forum.taichi.graphics/

Running example minimal ...
[Taichi] Starting on arch=x64
42.0
>>> Running time: 0.22s
42

Consider attaching this log when maintainers ask about system information.
>>> Running time: 5.42s
```

**Comment by ailzhang** [comment]
@aespielberg `vulkan` backend is not supported on Ubuntu16.04 so the wheel we released for 16.04 only supports cpu and cuda backend. If you want to use GGUI (or vulkan backend), try upgrading to ubuntu18.04+?

**Comment by aespielberg** [comment]
Ok, good to know.  It works fine on my Ubuntu 20,04 machine, I just didn't know if it wasn't supported for 16.04 or if it was something else (graphics card?).  I'll close the ticket.
